### PR TITLE
Adding translations for cosmos string methods

### PR DESCRIPTION
**Pull request by maumar**
INDEX_OF, LEFT, RIGHT, REPLACE, case insensitive STRINGEQUALS

### CODE REVIEW

File: src/EFCore.Cosmos/Query/Internal/StringMethodTranslator.cs [hunk 1]
```diff
@@ -212,6 +250,31 @@ public StringMethodTranslator(ISqlExpressionFactory sqlExpressionFactory)
                             arguments[3])));
             }
 
+            if (_stringComparisonWithComparisonTypeArgumentInstance.Equals(method)
+                || _stringComparisonWithComparisonTypeArgumentStatic.Equals(method))
+            {
+                var comparisonTypeArgument = arguments.Last();
```
Comment (smitpatel) [inline src/EFCore.Cosmos/Query/Internal/StringMethodTranslator.cs:256]: use indexer rather than Last.

**Comment by bricelam** [comment]
LENGTH is my adoption blocker :trollface:

**Comment by maumar** [comment]
@bricelam length is already in, did it my my previous pr: 7b4fe04182eb5b9cca7686a509acd4cb313ca5e1

File: src/EFCore.Cosmos/Query/Internal/StringMethodTranslator.cs [hunk 2]
```diff
@@ -1,12 +1,10 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Reflection;
 using Microsoft.EntityFrameworkCore.Diagnostics;
 using Microsoft.EntityFrameworkCore.Utilities;
+using Microsoft.VisualBasic;
```
Comment (smitpatel) [inline src/EFCore.Cosmos/Query/Internal/StringMethodTranslator.cs:7]: ?

File: src/EFCore.SqlServer/Query/Internal/SqlServerStringMethodTranslator.cs [hunk 3]
```diff
@@ -10,6 +10,7 @@
 using Microsoft.EntityFrameworkCore.Query;
 using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
 using Microsoft.EntityFrameworkCore.Utilities;
+using Microsoft.VisualBasic;
```
Comment (smitpatel) [inline src/EFCore.SqlServer/Query/Internal/SqlServerStringMethodTranslator.cs:13]: ?

**Comment by maumar** [comment]
vb got left and right methods so they map nicely to sql functions.

**Comment by maumar** [comment]
@smitpatel new version up

**Review by smitpatel — APPROVED** [review]

**Pull request merged**